pkg/kubernetes/statefulset.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"github.com/billiford/go-clouddriver/pkg/kubernetes/manifest"
+	v1 "k8s.io/api/apps/v1"
+)
+
+type StatefulSet interface {
+	Object() *v1.StatefulSet
+	SetReplicas(*int32)
+	Status() manifest.Status
+}
+
+type statefulSet struct {
+	ss *v1.StatefulSet
+}
+
+func NewStatefulSet(m map[string]interface{}) StatefulSet {
+	s := &v1.StatefulSet{}
+	b, _ := json.Marshal(m)
+	_ = json.Unmarshal(b, &s)
+
+	return &statefulSet{ss: s}
+}
+
+func (ss *statefulSet) Object() *v1.StatefulSet {
+	return ss.ss
+}
+
+func (ss *statefulSet) SetReplicas(replicas *int32) {
+	ss.ss.Spec.Replicas = replicas
+}
+
+func (ss *statefulSet) Status() manifest.Status {
+	s := manifest.DefaultStatus
+	x := ss.ss
+
+	updStrategyType := x.Spec.UpdateStrategy.Type
+	if updStrategyType == "OnDelete" {
+		return s
+	}
+
+	if x.ObjectMeta.Generation != x.Status.ObservedGeneration {
+		s.Stable.State = false
+		s.Stable.Message = "Waiting for Stateful Set spec update to be observed"
```

Comment:
make this match the expected message from the [source](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/UnstableReason.java#L25).

Reply:
Updated. Will commit once all requests implemented.